ssl/ssl_cert.c

```diff
@@ -1047,16 +1047,13 @@ static int ssl_security_default_callback(const SSL *s, const SSL_CTX *ctx,
         }
     case SSL_SECOP_VERSION:
         if (!SSL_IS_DTLS(s)) {
-            /* SSLv3 not allowed at level 2 */
-            if (nid <= SSL3_VERSION && level >= 2)
-                return 0;
-            /* TLS v1.1 and above only for level 3 */
-            if (nid <= TLS1_VERSION && level >= 3)
-                return 0;
-            /* TLS v1.2 only for level 4 and above */
-            if (nid <= TLS1_1_VERSION && level >= 4)
+            /* SSLv3, TLS v1.0 and TLS v1.1 only allowed at level 0 */
+            if (nid <= TLS1_1_VERSION && level > 0)
                 return 0;
         } else {
+            /* DTLS v1.0 only allowed at level 0 */
+            if (DTLS_VERSION_LE(nid, DTLS1_VERSION) && level > 0)
+                return 0;
             /* DTLS v1.2 only for level 4 and above */
             if (DTLS_VERSION_LT(nid, DTLS1_2_VERSION) && level >= 4)
                 return 0;
```

Comment:
There is no DTLSv1.1 so this `if` does not make any sense anymore.

Reply:
Oops....

Reply:
Fixed